book/src/dev/rfcs/0005-treestate.md

```diff
@@ -0,0 +1,193 @@
+# Treestate
+
+- Feature Name: treestate
+- Start Date: 2020-08-31
+- Design PR: [ZcashFoundation/zebra#983](https://github.com/ZcashFoundation/zebra/issues/983)
+- Zebra Issue: [ZcashFoundation/zebra#958](https://github.com/ZcashFoundation/zebra/issues/958)
+
+# Summary
+[summary]: #summary
+
+To validate blocks involving shielded transactions, we have to check the
+computed treestate from the included transactions against the block header
+metadata (for Sapling) or previously finalized state (for Sprout). This document
```

Comment:
Do we need to update this document for Orchard?

Reply:
Updated this section 👍 919491d48197fc3a19a879eefcfba45e0815877f